src/models/user.ts

```diff
@@ -0,0 +1,50 @@
+import mongoose from "mongoose";
+
+export interface IUser {
+    fullName: string;
+    email: string;
+    password: string;
+    role: string;
+    dateOfBirth: Date;
+    phoneNumber: string;
+}
+
+export interface IUserModel extends IUser, mongoose.Document { }
+
+const userSchema = new mongoose.Schema({
+    fullName: {
+        type: String,
+        required: true,
+    },
+    email: {
+        type: String,
+        required: true,
+        unique: true,
+        match: [
+            /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+            'Please provide a valid email',
+        ],
+    },
+    password: {
+        type: String,
+        required: true,
+    },
+    role: {
+        type: String,
+        required: true,
+        enum: ["admin", "patient", "doctor", "nurse"],
```

Comment:
Import this enum from another file such that if we change a specific user type we change it in a single place